### PR TITLE
Create chart snippet

### PR DESCRIPTION
**Pull request by RealOrangeOne**
Create a chart snipppet, for embedding into pages.

The only issue with this implementation is that the title on the chart itself isn't required, so sometimes the title in the list isn't shown. If present, the title is both shown in the list and used for search indexing.

[Screencast_20240716_114534.webm](https://github.com/user-attachments/assets/3416476e-ea80-435d-9e08-0156012a527a)

### CODE REVIEW

File: ons_alpha/core/models/snippets.py [hunk 1]
```diff
@@ -99,3 +102,25 @@ class Meta:
 
     def __str__(self):
         return str(self.name)
+
+
+@register_snippet
+class Chart(index.Indexed, PreviewableMixin, models.Model):
+    chart = StreamField([("chart", ChartBlock())], use_json_field=True, min_num=1, max_num=1)
```
Comment (RealOrangeOne) [inline ons_alpha/core/models/snippets.py:114]: I could swear there's a nicer way of doing this, I just forget what it is. For now, this does work, it just requires clicking an extra "+" to show the inputs.

File: ons_alpha/jinja2/templates/base.html [hunk 2]
```diff
@@ -28,30 +28,30 @@
...
-                        }
-                    ]
-                }
-            ],
-            "OGLLink": {
-                "pre": 'All content is available under the',
-                "link": 'Open Government Licence v3.0',
-                "url": 'https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/',
-                "post": ', except where otherwise stated'
+    },
```
Comment (RealOrangeOne) [inline ons_alpha/jinja2/templates/base.html:31]: There's no change here, just fixes the whitespace so the brackets match and it's easier to follow

File: ons_alpha/core/models/snippets.py [hunk 3]
```diff
@@ -99,3 +102,25 @@ class Meta:
...
+
+    panels = [FieldPanel("chart")]
+
+    search_fields = [index.SearchField("title")]
+
+    def __str__(self):
+        return str(self.title)
+
+    @property
+    def title(self):
```
Comment (zerolab) [inline ons_alpha/core/models/snippets.py]: suggestion: add a name field to use in admin listings. This way editors can give it an internal name

And use the chart title on the FE

File: ons_alpha/jinja2/templates/snippets/chart.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{% extends "layout/_template.njk" %}
+
+{% block main %}
+    {% include_block object.chart %}
+{% endblock %}
+
+{% block bodyEnd %}
```
Comment (zerolab) [inline ons_alpha/jinja2/templates/snippets/chart.html]: suggestion:
```suggestion
{% block scripts %}
    {{ super() }}
```

File: ons_alpha/core/models/snippets.py [hunk 5]
```diff
@@ -99,3 +102,25 @@ class Meta:
 
     def __str__(self):
         return str(self.name)
+
+
+@register_snippet
+class Chart(index.Indexed, PreviewableMixin, models.Model):
```
Comment (zerolab) [inline ons_alpha/core/models/snippets.py:109]: Could you please add it to `BulletinStoryBlock` and update the bulletin template to have the `include_django` include if we have a chart block?

File: ons_alpha/bulletins/blocks.py [hunk 6]
```diff
@@ -25,6 +26,7 @@ class BulletinStoryBlock(StreamBlock):
     embed = EmbedBlock()
     image = ImageChooserBlock()
     related_links = RelatedLinksBlock(RelatedContentBlock())
+    chart = SnippetChooserBlock("core.Chart", template="templates/components/streamfield/chart.html", icon="table")
```
Comment (zerolab) [inline ons_alpha/bulletins/blocks.py]: non-blocking: this is where https://github.com/wagtail-nest/wagtail-font-awesome-svg would come in handy

https://fontawesome.com/search?q=chart&o=r&m=free

**Review by MebinAbraham — COMMENTED** [review]
Can you add a quick screen cast that we can share with devs/UX/UR?.

File: ons_alpha/core/models/snippets.py [hunk 7]
```diff
@@ -99,3 +104,37 @@ class Meta:
...
+            return ""
+
+    def get_preview_template(self, request, mode_name):
+        return "templates/snippets/chart.html"
+
+
+class ChartViewSet(SnippetViewSet):
+    model = Chart
+    icon = "chart-simple"
+    list_display = ["name", "chart_title", UpdatedAtColumn()]
```
Comment (MebinAbraham) [inline ons_alpha/core/models/snippets.py:137]: Minor, would be nice to set `add_to_admin_menu = True`

File: ons_alpha/bulletins/blocks.py [hunk 8]
```diff
@@ -25,6 +26,7 @@ class BulletinStoryBlock(StreamBlock):
     embed = EmbedBlock()
     image = ImageChooserBlock()
     related_links = RelatedLinksBlock(RelatedContentBlock())
+    chart = ChartChooserBlock()
```
Comment (MebinAbraham) [inline ons_alpha/bulletins/blocks.py]: Nit: Add this under the `DataVis` group.

**Review by MebinAbraham — APPROVED** [review]
LGTM

**Review by zerolab — APPROVED** [review]

**Pull request merged**